google/cloud/storage/examples/storage_notification_samples.cc

```diff
@@ -74,12 +79,16 @@ void CreateNotification(google::cloud::storage::Client client, int& argc,
   //! [create notification] [START storage_create_pubsub_bucket_notification]
   namespace gcs = google::cloud::storage;
   [](gcs::Client client, std::string bucket_name, std::string topic_name) {
-    gcs::NotificationMetadata notification =
-        client
-            .CreateNotification(bucket_name, topic_name,
-                                gcs::payload_format::JsonApiV1(),
-                                gcs::NotificationMetadata())
-            .value();
+    gcs::StatusOr<gcs::NotificationMetadata> status = client.CreateNotification(
+        bucket_name, topic_name, gcs::payload_format::JsonApiV1(),
+        gcs::NotificationMetadata());
+    if (not status.ok()) {
+      std::cerr << "Error creating notification for " << bucket_name
+                << " on topic " << topic_name << ", status=" << status.status()
+                << std::endl;
+      return;
+    }
+    gcs::NotificationMetadata notification = std::move(*status);
```

Comment:
Rather than std::moving into a new object, I think we may want to encourage using StatusOr as a smart pointer. It results in efficient code, fewer copies/moves, and it looks similar to unique_ptr and optional usage.

If we did this, I'd suggest that the name of the object returned from CreateNotification should probably be 'notification' (just like a unique_ptr<Foo> is probably named 'foo', not 'up'). Then the code might look something like:

```cc
gcs::StatusOr<gcs::NotificationMetadata> notification = client.CreateNotification(...);
if (!notification) {
  ...
}
std::cout << ... << notification->id() << ... << *notification << '\n';
```

Reply:
LGTM! Thanks @devjgm, looking at the example `notification->id()` accessing the value does make this example much easier to read. 

Accessing at least one metadata field helps me understand the example when using the type `StatusOr<>`.  @coryan could you update the examples to at least access one metadata field when dumping all the metadata fields similar to this example? It would help.